drivers/clock_control/clock_control_ti_syscon.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2025 Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_AM654_COMPAT ti_am654_ehrpwm_tbclk
+#define DT_AM64_COMPAT  ti_am64_epwm_tbclk
+#define DT_AM62_COMPAT  ti_am62_epwm_tbclk
+
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/syscon.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ti_syscon_gate_clk, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#define DEV_CFG(dev) ((struct ti_syscon_gate_clk_cfg *)(dev)->config)
+
+struct ti_syscon_gate_clk_id_data {
+	uint32_t offset;
+	uint32_t bit;
+};
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_AM64_COMPAT)
+static const struct ti_syscon_gate_clk_id_data am64_clk_ids[] = {
+	{0, 0}, {0, 1}, {0, 2}, {0, 3}, {0, 4}, {0, 5}, {0, 6}, {0, 7}, {0, 8},
```

Comment:
Any good way to encode this data into DT so that we don't need to keep device specific data here in the driver?